### PR TITLE
Port func_vehicle implementation from cs16-client/regamedll_cs.

### PR DESCRIPTION
**Pull request by nekonomicon**

### CODE REVIEW

File: dlls/vehicle.cpp [hunk 1]
```diff
@@ -0,0 +1,996 @@
+#include "extdll.h"
```
Comment (a1batross) [inline dlls/vehicle.cpp:1]: Maybe credit ReGameDLL_CS developers here?

Comment (nekonomicon) [reply dlls/vehicle.cpp:1]: I'm not sure how it may look.

**Comment by a1batross** [comment]
Yeah, they don't have anything on this file...

**Comment by FreeSlave** [comment]
I tried the branch with this map https://twhl.info/vault/view/6815
I can't turn the vehicle for some reason, only move forward.
Also the sound doesn't stop after I stop the vehicle, but it behaves so in HL as well.

And regamedll claims to be GPL so I dunno how it's gonna work with hlsdk license.

**Comment by FreeSlave** [comment]
Found the proble with turnings: https://github.com/FWGS/hlsdk-portable/blob/2aebf12b1f18cbd695765c918bb36b1341783cd4/dlls/player.cpp#L1905
If stops train control when player strafes.

**Comment by nekonomicon** [comment]
@FreeSlave try the latest commit.

> And regamedll claims to be GPL so I dunno how it's gonna work with hlsdk license.

It has exception, but IMHO It's very unclear.
Anyway we can modify this code when valve released SDK update.

**Comment by FreeSlave** [comment]
Turn controls work fine after the latest change

**Comment by N7P0L3ON** [comment]
@nekonomicon any idea how to properly credit the player when he runs over / kills someone? The current implementation just takes 1 frag out of the one driving the func_vehicle for each kill... that might be desirable if you have it in Counter Strike and you're running over a teammate.. but what about if we push this to Half-Life? There it's free for all... and maybe for shits and giggles you need to get a point for fragging someone say with a forklift or other vehicle?

**Comment by nekonomicon** [comment]
@N7P0L3ON Why not ask Valve about that?
Also, Half-Life has teamplay gamemode.

**Comment by N7P0L3ON** [comment]
@nekonomicon Thanks for the good suggestion. We got it fixed already.

**Pull request merged**